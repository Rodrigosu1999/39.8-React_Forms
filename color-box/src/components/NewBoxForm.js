import React, {useState} from "react";

const NewBoxForm = ({addBox}) => {
    const INITIAL_STATE = {
        color: "#FF0000",
        height: 100,
        width: 100 
    }
    const [formData, setFormData] = useState(INITIAL_STATE);

    const handleChange = (e) => {
        const {name, value} = e.target;
        setFormData(formData => ({
            ...formData,
            [name]: value
        }));
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        addBox({...formData});
        setFormData(INITIAL_STATE);
    }

    return (
        <form onSubmit={handleSubmit}>
            <label htmlFor="color">Color</label>
            <input
                id="color"
                name="color" 
                type="color" 
                value={formData.color}
                onChange={handleChange} 
            />
            <label htmlFor="height">Height</label>
            <input
                id="height"
                name="height" 
                type="range" 
                min="20"
                max="200"
                value={formData.height}
                onChange={handleChange} 
            />
            <label htmlFor="width">Width</label>
            <input
                id="width"
                name="width" 
                type="range" 
                min="20"
                max="200"
                value={formData.width}
                onChange={handleChange} 
            />
            <button>Add new box!</button>
        </form>
    );
}

export default NewBoxForm;