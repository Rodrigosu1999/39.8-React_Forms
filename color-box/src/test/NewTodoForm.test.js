import React from 'react';
import { getByText, render, screen } from '@testing-library/react';
import NewBoxForm from '../components/NewBoxForm';

it('Renders App', () => {
  render(<NewBoxForm />);
});

it('Passes Snapshot', () => {
  const {asFragment} = render(<NewBoxForm />);
  expect(asFragment()).toMatchSnapshot();
});

it('renders correct text', () => {
  const {getByText} = render(<NewBoxForm />);
  console.log(getByText('Add new box!'))
});
