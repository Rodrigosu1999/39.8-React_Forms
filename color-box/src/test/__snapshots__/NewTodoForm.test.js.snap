// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Passes Snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="color"
    >
      Color
    </label>
    <input
      id="color"
      name="color"
      type="color"
      value="#FF0000"
    />
    <label
      for="height"
    >
      Height
    </label>
    <input
      id="height"
      max="200"
      min="20"
      name="height"
      type="range"
      value="100"
    />
    <label
      for="width"
    >
      Width
    </label>
    <input
      id="width"
      max="200"
      min="20"
      name="width"
      type="range"
      value="100"
    />
    <button>
      Add new box!
    </button>
  </form>
</DocumentFragment>
`;
